src/common/connection.ts

```diff
@@ -32,10 +34,84 @@ export interface ConnectionOptions {
 /**
  * ConnectionUserOptions is the user-provided configuration object. All configuration
  * is optional, so any ConnectionOptions configuration that has a default value is
- * still optional for the user to provide.
+ * still optional at the point that the user provides it.
  */
 export type ConnectionUserOptions = Partial<ConnectionOptions>
 
+/**
+ * Represents an intentionally triggered web-socket disconnect code.
+ * WebSocket spec allows 4xxx codes for app/library specific codes.
+ * See: https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent
+ **/
+const INTENTIONAL_DISCONNECT_CODE = 4000
+
+/**
+ * Create a new websocket given your URL and optional proxy/certificate
+ * configuration.
+ */
+function createWebSocket(url: string, config: ConnectionOptions): WebSocket {
```

Comment:
Moved off of the Connection class (no changes to the logic itself besides updating the references to use function arguments instead of `this`).